lms/djangoapps/experiments/views.py

```diff
@@ -1,26 +1,37 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
 from edx_rest_framework_extensions.authentication import JwtAuthentication
 from rest_framework import permissions, viewsets
+from rest_framework.decorators import list_route
 from rest_framework.filters import DjangoFilterBackend
+from rest_framework.response import Response
 
-from experiments import filters
-from experiments.models import ExperimentData
-from experiments.permissions import IsStaffOrOwner
-from experiments.serializers import ExperimentDataSerializer
+from experiments import filters, serializers
+from experiments.models import ExperimentData, ExperimentKeyValue
+from experiments.permissions import IsStaffOrOwner, IsStaffOrReadOnly
 from openedx.core.lib.api.authentication import SessionAuthenticationAllowInactiveUser
 
+User = get_user_model()  # pylint: disable=invalid-name
+
 
 class ExperimentDataViewSet(viewsets.ModelViewSet):
     authentication_classes = (JwtAuthentication, SessionAuthenticationAllowInactiveUser,)
     filter_backends = (DjangoFilterBackend,)
     filter_class = filters.ExperimentDataFilter
     permission_classes = (permissions.IsAuthenticated, IsStaffOrOwner,)
     queryset = ExperimentData.objects.all()
-    serializer_class = ExperimentDataSerializer
+    serializer_class = serializers.ExperimentDataSerializer
+    _cached_users = {}
 
     def filter_queryset(self, queryset):
         queryset = queryset.filter(user=self.request.user)
```

Comment:
@nasthagiri this is where we filter `ExperimentData` objects for the current user.